cmd/server.go

```diff
@@ -45,46 +45,31 @@ import (
 	config "github.com/k0sproject/k0s/pkg/apis/v1beta1"
 )
 
-// ServerCommand ...
-func ServerCommand() *cli.Command {
-	return &cli.Command{
-		Name:   "server",
-		Usage:  "Run server",
-		Action: startServer,
-		Flags: []cli.Flag{
-			&cli.StringFlag{
-				Name:    "config",
-				Aliases: []string{"c"},
-				Value:   "k0s.yaml",
-			},
-			&cli.BoolFlag{
-				Name:  "enable-worker",
-				Value: false,
-			},
-			&cli.StringFlag{
-				Name:  "profile",
-				Value: "default",
-				Usage: "worker profile to use on the node",
-			},
-		},
-		ArgsUsage: "[join-token]",
-	}
+func init() {
+	serverCmd.Flags().StringVar(&serverProfile, "profile", "default", "worker profile to use on the node")
+	serverCmd.Flags().BoolVar(&enableWorker, "enable-worker", false, "enable worker (default false)")
 }
 
-func configFromCmdFlag(ctx *cli.Context) (*config.ClusterConfig, error) {
-	clusterConfig := ConfigFromYaml(ctx)
-
-	errors := clusterConfig.Validate()
-	if len(errors) > 0 {
-		messages := make([]string, len(errors))
-		for _, e := range errors {
-			messages = append(messages, e.Error())
-		}
-		return nil, fmt.Errorf("config yaml does not pass validation, following errors found:%s", strings.Join(messages, "\n"))
+var (
+	serverProfile string
```

Comment:
should be `workerProfile`, also need to be changed for the flag

Reply:
changed to `serverWorkerProfile` so as to not collide with `workerProfile` in `worker.go`